python_ta/checkers/redundant_assignment_checker.py

```diff
@@ -83,12 +82,13 @@ def _transfer(self, block: CFGBlock, out_facts: Set[str], scope: Union[astroid.M
         gen = out_facts.copy()
         kill = set()
         for statement in reversed(block.statements):
+            if isinstance(statement, astroid.FunctionDef):
```

Comment:
This check is added because the `nodes_of_class` method in the `for` loop below will look into function definitions for the required nodes, which is not what it wanted. The new test case passes because of this specific check.

Reply:
Okay this is good. Please add a comment in the code.